core/ui/src/main/java/com/yapp/core/ui/extension/UriHandler.kt

```
@@ -0,0 +1,20 @@
+import timber.log.Timber
+
+fun UriHandler.safeOpenUri(rawUrl: String) {
+    val url = if (!rawUrl.startsWith("http://") && !rawUrl.startsWith("https://")) {
+        "https://$rawUrl"
+    } else {
+        rawUrl
+    }
+
+    try {
+        openUri(url)
+    } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)